From 010a99cb725c033772d595838d2cfb282eb7c94b Mon Sep 17 00:00:00 2001
From: John Clements <John.Clements@amd.com>
Date: Thu, 25 Jul 2019 15:29:11 +0800
Subject: [PATCH 3262/4256] drm/amdgpu: update PSP CMD fail response status
 print

Print the response in hex with the apprpriate mask.

Change-Id: I83f1a9147778fdcc6e2d246b4b6daf2a857c75a0
Signed-off-by: John Clements <John.Clements@amd.com>
Reviewed-by: Hawking Zhang <Hawking.Zhang@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c
index 1c0a1e2dba78..0c0f3cb2b627 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c
@@ -167,8 +167,8 @@ psp_cmd_submit_buf(struct psp_context *psp,
                 if (ucode)
                         DRM_WARN("failed to load ucode id (%d) ",
                                   ucode->ucode_id);
-                DRM_WARN("psp command failed and response status is (%d)\n",
-                          psp->cmd_buf_mem->resp.status);
+		DRM_WARN("psp command failed and response status is (0x%X)\n",
+				psp->cmd_buf_mem->resp.status & GFX_CMD_STATUS_MASK);
 		if (!timeout) {
 			mutex_unlock(&psp->mutex);
                 	return -EINVAL;
-- 
2.17.1

